Added Rest-Controllers, -Models and -Repositories for Evidence-Formulas.

fabiom6 opened the pull request.

Commit:
Added Rest-Controllers, -Models and -Repositories for Evidence-Formulas.

Commit:
Fixed a bug while updating evidence-formula with invalid name.

Updating an evidence-formula with an invalid new name no longer receives a 500 "Internal Server Error" response.

DevLeoko commented on backend/src/main/java/io/dcbn/backend/evidenceFormula/controller/EvidenceFormulaController.java:
```diff
@@ -0,0 +1,95 @@
+  public ResponseEntity<Void> updateEvidenceFormulaByName(@PathVariable String name,
+      @Valid @RequestBody EvidenceFormula evidenceFormula) {
+    if (repository.existsByName(evidenceFormula.getName())) {
+      return ResponseEntity.badRequest().build();
```
Dann failed der request aber wenn man zB nur die Formel updated und nicht den Namen oder?

fabiom6 replied:
Das stimmt auch...
Ich hab ein bisschen damit rumgetestet und musste feststellen, dass Hibernate es nicht mag wenn man die ID eines Entities ändert. Spricht was dagegen eine ID einzufügen und darüber auch alle Anfragen zu machen? Der Name der Formel kann ja trotzdem eindeutig sein.

DevLeoko commented on backend/src/main/java/io/dcbn/backend/evidenceFormula/controller/EvidenceFormulaController.java:
```diff
@@ -0,0 +1,95 @@
+    if (repository.existsByName(evidenceFormula.getName())) {
+      return ResponseEntity.badRequest().build();
+    }
+    if (Strings.isNullOrEmpty(evidenceFormula.getName())) {
```
Das sollte eher in der Evid.Formel validiert werden oder? Also mit sowas wie @min und @max

fabiom6 replied:
Das wird auch schon gemacht.
Ich Genie hab nur vergessen das aus dem Controller zu löschen.

DevLeoko commented on backend/src/main/java/io/dcbn/backend/evidenceFormula/controller/EvidenceFormulaController.java:
```diff
@@ -0,0 +1,95 @@
+  @PreAuthorize("hasRole('ADMIN')")
+  public ResponseEntity<EvidenceFormula> getEvidenceFormulaByName(@PathVariable String name) {
+    Optional<EvidenceFormula> evidenceFormula = repository.findById(name);
+    return evidenceFormula.map(ResponseEntity::ok).orElse(ResponseEntity.notFound().build());
```
Props für den kranken Java 8 code ;D

Commit:
Fixed bug when updating evidence-formula.

Updating a evidence-formula no longer fails when not changing the formula name.

fabiom6 commented:
Ich hab jetzt mal die von mir vorgeschlagenen Änderungen committed und den Fehler beim Updaten (hoffentlich) gefixed.
